packages/requisitions/src/RequestRequisition/api/hooks.ts

```diff
@@ -202,3 +204,56 @@ export const useDeleteSelectedRequisitions = () => {
 
   return deleteAction;
 };
+
+export const useAddFromMasterList = () => {
+  const { error } = useNotification();
+  const queryClient = useQueryClient();
+  const { id: requestId, requisitionNumber } = useRequestFields([
+    'id',
+    'requisitionNumber',
+  ]);
+  const api = useRequestApi();
+  const mutationState = useMutation(api.addFromMasterList, {
+    onSettled: () =>
+      queryClient.invalidateQueries(api.keys.detail(String(requisitionNumber))),
+  });
```

Comment:
I'm thinking that all invalidates can pretty much be on the `onSettled` rather than `onSuccess`. Reasoning is because the rejection may be because something has happened to the record on the server since queried.. could invalidate on a specific error which might be caused by this use case, but I think it's simpler to just *always* use `onSettled` to avoid forgetting